src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,203 @@
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarWithManufacturerFromResultSet(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars from DB", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        deleteRelationsFromCarsAndDrivers(car);
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            insertDriver(car);
+            statement.executeUpdate();
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car: " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by ID: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverID) {
+        String query = "SELECT c.id, model, manufacturer_id, name, country "
+                + "FROM CARS c "
+                + "JOIN MANUFACTURERS m "
+                + "ON c.manufacturer_id = m.id "
+                + "JOIN CARS_DRIVERS cd "
+                + "ON c.id = cd.car_id "
+                + "WHERE c.is_deleted = FALSE AND driver_id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverID);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarWithManufacturerFromResultSet(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars from DB", e);
+        }
+    }
+
+    private void deleteRelationsFromCarsAndDrivers(Car car) {
```

Comment:
I suggest it's better to change the name on something more specific cause you are deleting drivers for car by specific ID, and it's not obvious from the name for now